### PR TITLE
2022_10_10

### PR DESCRIPTION
**Pull request by Kukuru-Dza**
написал, но почему то не запускается тест

### CODE REVIEW

File: first_lessons_dushanbe/lesson_array.py [hunk 1]
```diff
@@ -1,12 +1,15 @@
...
+from define_count_and_index import fn_index
+
+
 def find_number(a, b):
     if a is None or b is None:
         return -1
 
-    if a.count(b) > 0:
-        return a.index(b)
+    if fn_index(a, b):
```
Comment (shuvava) [inline first_lessons_dushanbe/lesson_array.py]: you does not need to call `fn_index` twice. you can you it once.

File: first_lessons_dushanbe/array_sorting.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+
+def list_sort(a):
+    if a is None:
+        return False
```
Comment (shuvava) [inline first_lessons_dushanbe/array_sorting.py]: you should not change return type depending on input conditions. Could you please make return type consistent?

File: first_lessons_dushanbe/array_sorting.py [hunk 3]
```diff
@@ -0,0 +1,22 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+
+def list_sort(a):
```
Comment (shuvava) [inline first_lessons_dushanbe/array_sorting.py:4]: Could you please add comments with operation complexity for this method?

Comment (shuvava) [inline first_lessons_dushanbe/array_sorting.py:4]: could you please try to implement sorting in-place?

File: first_lessons_dushanbe/array_sorting.py (hunk 2 above)
Comment (Kukuru-Dza) [reply first_lessons_dushanbe/array_sorting.py]: Yes, I've changed it to return a

File: first_lessons_dushanbe/array_sorting.py (hunk 3 above)
Comment (Kukuru-Dza) [reply first_lessons_dushanbe/array_sorting.py:4]: done

Comment (Kukuru-Dza) [reply first_lessons_dushanbe/array_sorting.py:4]: array_sorting_one_array

but for me it was much more complicated to do it and I'm not sure that this method is less resource consuming.

File: first_lessons_dushanbe/lesson_array.py (hunk 1 above)
Comment (Kukuru-Dza) [reply first_lessons_dushanbe/lesson_array.py]: x = fn_index(a, b)
    if x >= 0:
        return x
    else:
        return -1

is it OK? or I'm thinking wrong?

File: first_lessons_dushanbe/array_sorting_one_array.py [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    count = 0
+    while count < len(a)-1:
+        count1 = count
+        min_val = a[count1]
+        while count1 < (len(a) - 1):
+            if a[count1 + 1] < min_val:
+                min_val = a[count1 + 1]
+                ind_min_val = count1 + 1
+            count1 += 1
+        a.remove(a[ind_min_val])
```
Comment (shuvava) [inline first_lessons_dushanbe/array_sorting_one_array.py:17]: can you implement it with replace operation?

File: first_lessons_dushanbe/lesson_array.py (hunk 1 above)
Comment (shuvava) [reply first_lessons_dushanbe/lesson_array.py]: yes it is correct

**Comment by shuvava** [comment]
Great work!!!

**Pull request merged**